Add meteor build timeout option and pass it to BuildAsPromise.

marekrogala opened the pull request.

apendua commented on bin/gagarin:
```diff
@@ -36,6 +36,7 @@ program
   .option('-m, --mute-build', 'do not show build logs', false)
   .option('-f, --flavor <name>', 'default flavor of api (promise, fiber)', 'promise')
   .option('-T, --startup-timeout <ms>' ,'server startup timeout [5000]', intParse, 5000)
+  .option('-U, --meteor-startup-timeout <ms>' ,'initial meteor startup timeout [120000]', intParse, 120000);
```
@marekrogala I am wondering why do we need another option here. Can't we just use `startupTimeout`? If the reason is that builds would usually take a little more time to complete then why not use `buildTimeout` instead? I guess it would be a lot more intuitive.

apendua commented:
@marekrogala Good catch! Thank you for this update. I only have objections to the naming convention, which I described in a comment.

marekrogala commented:
@apendua Thanks. I agree with the naming suggestion, updated. Also fixed a typo in an error message.

I didn't use ```startupTimeout``` since it has a default value of ```5000``` so this change would significantly decrease the default timeout for BuildPromise, since previously a hardcoded value of ```120000``` was used.

Another option would be to use ```startupTimeout``` and increase its default value to e.g. 120000, but I'm not sure if its OK to do that, since it is also used in other places?

The change itself is motivated by our continuous integration (Wercker), which is much slower than our dev machines and makes gagarin exceed this limit often.

apendua commented:
@marekrogala Ok, thanks for the explanation. I can see that adding this option is totally justified. This will probably end up in the next release at the beginning of the next week. I am glad you have chosen to use Gagarin for your project. Please not that it's still in a pretty early development stage so if there are any problems you encounter, please don't hesitate to report them.
